android/src/main/java/expo/modules/xmtpreactnativesdk/XMTPModule.kt

```diff
@@ -1153,15 +1196,15 @@ class XMTPModule : Module() {
     ): Group? {
         val client = clients[clientAddress] ?: throw XMTPException("No client")
 
-        val cacheKey = "${clientAddress}:${id}"
+        val cacheKey = "${client.inboxId}:${id}"
         val cacheGroup = groups[cacheKey]
         if (cacheGroup != null) {
             return cacheGroup
         } else {
             val group = client.conversations.listGroups()
                 .firstOrNull { it.id.toHex() == id }
             if (group != null) {
-                groups[group.cacheKey(clientAddress)] = group
+                groups[group.cacheKey(client.inboxId)] = group
```

Comment:
Think this needs to be updated on line 516 as well:

https://github.com/xmtp/xmtp-react-native/blob/3542e807261f13a186be9cd68d414fa5ae1acccb/android/src/main/java/expo/modules/xmtpreactnativesdk/XMTPModule.kt#L516